File: src/main/java/arcana/Arcana.java
package arcana;

import arcana.common.ArcanaGroup;
import net.minecraftforge.fml.common.Mod;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

@Mod(Arcana.id)
public class Arcana {
    public static final String id = "arcana";
    public static final Logger logger = LogManager.getLogger(id);
    public static ArcanaGroup ARCANAGROUP = new ArcanaGroup();

    public Arcana(){
    }

}

File: src/main/kotlin/arcana/Arcana.kt
package arcana

import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.config.ModConfig
import org.apache.logging.log4j.LogManager
import thedarkcolour.kotlinforforge.forge.LOADING_CONTEXT

@Mod(Arcana.id)
object Arcana {
    const val id = "arcana"
    val logger = LogManager.getLogger(id)
    init {
        //must be registered before any events
        LOADING_CONTEXT.registerConfig(ModConfig.Type.COMMON, ArcanaConfig.COMMON_SPEC)
        LOADING_CONTEXT.registerConfig(ModConfig.Type.CLIENT, ArcanaConfig.CLIENT_SPEC)
    }
}